x/cronos/keeper/evm.go

```diff
@@ -95,47 +94,84 @@ func (k Keeper) ConvertCoinFromNativeToCRC20(ctx sdk.Context, sender common.Addr
 
 		k.Logger(ctx).Info(fmt.Sprintf("contract address %s created for coin denom %s", contract.String(), coin.Denom))
 	}
-	err = k.bankKeeper.SendCoins(ctx, sdk.AccAddress(sender.Bytes()), sdk.AccAddress(contract.Bytes()), sdk.NewCoins(coin))
-	if err != nil {
-		return err
-	}
-	_, err = k.CallModuleCRC20(ctx, contract, "mint_by_cronos_module", sender, coin.Amount.BigInt())
-	if err != nil {
-		return err
+
+	isSource := types.IsSourceCoin(coin.Denom)
+	if isSource {
+		// burn coins
+		err = k.bankKeeper.SendCoinsFromAccountToModule(ctx, sdk.AccAddress(sender.Bytes()), types.ModuleName, sdk.NewCoins(coin))
+		if err != nil {
+			return err
+		}
+		err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, sdk.NewCoins(coin))
+		if err != nil {
+			return err
+		}
+		// unlock crc tokens
+		_, err = k.CallModuleCRC21(ctx, contract, "transfer_by_cronos_module", sender, coin.Amount.BigInt())
```

Comment:
```solidity
    function transfer_by_cronos_module(address addr, uint amount) public {
        require(msg.sender == module_address);
        unsafe_transfer(addr, module_address, amount);
    }
```
it seems `transfer_by_cronos_module` is used to lock crc21 tokens?

Reply:
should be `transfer_to_cronos_module`?

Reply:
I mean the comment says `unlock crc tokens`, so I think we should unlock tokens by calling `transfer_from_cronos_module`.

Reply:
yes it should be `transfer_from_cronos_module`